### DISCUSSION TITLE
wire:loading.attr="disabled" not working with wire:click

### DISCUSSION THREAD

**Discussion by AdamEsterle** [livewire/livewire#4755] [7. Bugs (archived)] [answered]
Taken from documentation here:
https://laravel-livewire.com/docs/2.x/loading-states#toggling-attributes

When doing a file upload, with "WithFileUpload", the below code will NOT disable the Cancel button

```
<div>
    <button wire:click="cancel" wire:loading.attr="disabled">
        Cancel
    </button>
</div>
```

However, if you take off the
`wire:click="cancel"`
then this code WILL WORK and disable the Cancel button when a file is uploading
```
<div>
    <button wire:loading.attr="disabled">
        Cancel
    </button>
</div>
```

**Comment by silnarm** [comment, marked as answer]
You're not giving it a `wire:target`, so without the `wire:click` it is crawling up the DOM and finding the first sensible thing to attach itself to, which happens to be your upload. With `wire:click` on it, it becomes a valid target itself and that same DOM search ends quickly and with different results, the `wire:loading` only applies to your cancel action.

You will need to put a `wire:target` in there, referencing your upload (the thing to attach the loading directive to).

See: https://laravel-livewire.com/docs/2.x/loading-states#targeting-actions

**Reply by AdamEsterle** [reply]
That was it! Thanks a bunch

**Reply by EriksonEnterprises** [reply]
I know this is old but thanks!

Also for anyone coming around later, you can always just make up a target and use:

`wire:target.except="null"`

When doing that, it'll fire the attr for basically everything like loading state, and other events (which is what I wanted lol)
